src/goto-symex/goto_state.cpp

```diff
@@ -34,3 +36,59 @@ std::size_t goto_statet::increase_generation(
 
   return current_emplace_res.first->second.second;
 }
+
+/// Given a condition that must hold on this path, propagate as much knowledge
+/// as possible. For example, if the condition is (x == 5), whether that's an
+/// assumption or a GOTO condition that we just passed through, we can propagate
+/// the constant '5' for future 'x' references. If the condition is (y == &o1)
+/// then we can use that to populate the value set.
+/// \param condition: condition that must hold on this path
```

Comment:
Should you mention that this should be L2? There's a short end-of-line comment to that effect in the declaration.

Reply:
Done